registration-processor/init/registration-processor-packet-receiver-stage/src/main/java/io/mosip/registration/processor/packet/receiver/service/impl/PacketReceiverServiceImpl.java

```diff
@@ -243,10 +243,11 @@ private void validatePacketWithSync(SyncRegistrationEntity regEntity, String reg
 	 * @throws IOException
 	 *             Signals that an I/O exception has occurred.
 	 */
-	private boolean storePacket(String packetId, String stageName, SyncRegistrationEntity regEntity, InternalRegistrationStatusDto dto,
-			LogDescription description) throws IOException {
+	private boolean storePacket(String stageName, SyncRegistrationEntity regEntity, InternalRegistrationStatusDto dto,
+			LogDescription description) {
 		Boolean storageFlag = false;
-		dto = registrationStatusService.getRegistrationStatus(regEntity.getRegistrationId(), regEntity.getRegistrationType(), null);
+		dto = registrationStatusService.getRegistrationStatus(regEntity.getRegistrationId(),
+				regEntity.getRegistrationType(), 1, regEntity.getWorkflowInstanceId());
```

Comment:
calling method getIterationForSyncRecord()

Reply:
I see the same method getIterationForSyncRecord(regEntity) is called twice in the same method, can we store the value in a variable and reuse ? This is to avoid 2 dips to DB